server/utilities/helperFunctions.ts

```diff
@@ -152,3 +152,122 @@ export function determineLoginId(auth) {
     const decodedToken = jwt.decode(auth, { json: true });
     return parseInt(decodedToken.sub.split('|')[1], 10);
 }
+
+export function timeToMilliSeconds(time: string) {
+    const [hourMin, period] = time.split(' ');
+    const [hour, min] = hourMin.split(':');
+    const convertedHour = hour === '12' ? 3600000 : 3600000 * parseInt(hour, 10);
+    const convertedMin = 60000 * parseInt(min, 10);
+    const convertedPeriod = period === 'AM' ? 0 : 43200000;
+
+    return convertedHour + convertedMin + convertedPeriod;
+}
+
+export function isInTime(target: string, start: string, end: string): boolean {
+    const targetTime = timeToMilliSeconds(target);
+    const startTime = timeToMilliSeconds(start);
+    const endTime = timeToMilliSeconds(end);
+    return startTime <= targetTime && targetTime <= endTime;
+}
+
+export function isTimeBefore(comparing: string, target: string): boolean {
+    const targetTime = timeToMilliSeconds(target);
+    const comparingTime = timeToMilliSeconds(comparing);
+    return comparingTime < targetTime;
+}
+
+// Recursively checks if the order is correct by checking from last element to first.
+// eslint-disable-next-line consistent-return
+export function correctOrder(arr, lastIdx, target, type) {
+    if (lastIdx === -1) {
+        arr.unshift(target);
+        return arr;
+    }
+    const condition =
+        type === 'order' ? target.order < arr[lastIdx].order : !isTimeBefore(arr[lastIdx].time, target.time);
+    if (!condition) {
+        arr.splice(lastIdx + 1, 0, target);
+        return arr;
+    }
+    if (condition) return correctOrder(arr, lastIdx - 1, target, type);
```

Comment:
You might not need the eslint-disable if you combine the exclusive conditions:
```suggestion
    if (!condition) {
        arr.splice(lastIdx + 1, 0, target);
        return arr;
    } else {
        return correctOrder(arr, lastIdx - 1, target, type);
    }
```

Reply:
agreed. one thing about your suggestion is that airbnb eslint (which is industry standard) doesn't like 
`else return` saying that else is redundant

so I'll be making it
```
  if (!condition) {
        arr.splice(lastIdx + 1, 0, target);
        return arr;
    }
    return correctOrder(arr, lastIdx - 1, target, type);
```

Reply:
yup sounds right